latency/src/bin/z_ping.rs

```diff
@@ -14,58 +14,61 @@
 use async_std::stream::StreamExt;
 use async_std::sync::{Arc, Barrier, Mutex};
 use async_std::task;
+use clap::Parser;
 use std::collections::HashMap;
-use std::convert::TryInto;
 use std::time::{Duration, Instant};
-use structopt::StructOpt;
-use zenoh::*;
+use zenoh::config::Config;
+use zenoh::net::protocol::core::WhatAmI;
+use zenoh::prelude::*;
+use zenoh::publication::CongestionControl;
 
-#[derive(Debug, StructOpt)]
-#[structopt(name = "z_ping")]
+#[derive(Debug, Parser)]
+#[clap(name = "z_ping")]
 struct Opt {
-    #[structopt(short = "l", long = "locator")]
+    #[clap(short, long)]
     locator: Option<String>,
-    #[structopt(short = "m", long = "mode")]
+    #[clap(short, long)]
     mode: String,
-    #[structopt(short = "p", long = "payload")]
+    #[clap(short, long)]
     payload: usize,
-    #[structopt(short = "n", long = "name")]
+    #[clap(short, long)]
     name: String,
-    #[structopt(short = "s", long = "scenario")]
+    #[clap(short, long)]
     scenario: String,
-    #[structopt(short = "i", long = "interval")]
+    #[clap(short, long)]
     interval: f64,
-    #[structopt(long = "parallel")]
+    #[clap(long = "parallel")]
     parallel: bool,
 }
 
-async fn parallel(opt: Opt, config: Properties) {
-    let zenoh = Zenoh::new(config.into()).await.unwrap();
-    let zenoh = Arc::new(zenoh);
+async fn parallel(opt: Opt, config: Config) {
+    let session = zenoh::open(config).await.unwrap();
+    let session = Arc::new(session);
 
     // The hashmap with the pings
     let pending = Arc::new(Mutex::new(HashMap::<u64, Instant>::new()));
     let barrier = Arc::new(Barrier::new(2));
 
     let c_pending = pending.clone();
     let c_barrier = barrier.clone();
-    let c_zenoh = zenoh.clone();
+    let c_session = session.clone();
     let scenario = opt.scenario;
     let name = opt.name;
     let interval = opt.interval;
     task::spawn(async move {
-        let workspace = c_zenoh.workspace(None).await.unwrap();
-        let mut sub = workspace
-            .subscribe(&"/test/pong/".to_string().try_into().unwrap())
+        let mut sub = c_session
+            .subscribe("/test/pong/")
+            .reliable() // Default of the reliability is `best_effort`
             .await
             .unwrap();
 
         // Notify that the subscriber has been created
         c_barrier.wait().await;
 
-        while let Some(change) = sub.next().await {
```

Comment:
I would suggest preserver the better `while let` writing rather than `loop`. It's more common in Rust.

Reply:
Ok, I will change  it to `while let`, thanks.